instance.tf

```diff
@@ -0,0 +1,53 @@
+data "aws_ami" "amazonlinux" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-kernel-5.10-hvm-2.0.20220805.0-x86_64-gp2"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["137112412989"] # Canonical
+}
+
+resource "aws_instance" "public" {
+  ami                         = data.aws_ami.amazonlinux.id
+  instance_type               = "t2.micro"
+  associate_public_ip_address = true
+  key_name                    = "mykey"
```

Comment:
Please change the key name to "main"
It will then be easier for me to debug your code